### PR TITLE
fix(theme): allow a csp nonce to be specified

### LINKED ISSUE #15358: [Feature Request] [3.0.0-beta.4] CSP Nonce
### Problem to solve
A content security policy currently prevents usage of Vuetify v3, since it blocks unsafe-inline styles.
By providing a nonce it is possible to use Vuetify without lowering the protection of the CSP.

### Proposed solution
Same solution that is already implemented on Vuetify v2 (https://vuetifyjs.com/en/features/theme/#csp-nonce)

### PR DESCRIPTION
**Pull request by hendric-dev**
## Description

Allows to set a CSP Nonce via options on Vuetify v3
resolves #15358

## Motivation and Context

A content-security-policy is an effective mechanism to improve the security of a web application. Vuetify's theming is added as an inline script, which is not allowed by most CSPs.  
In order to make it pass the policy, it is necessary to add a nonce to the script and style sources.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src#unsafe_inline_script

## How Has This Been Tested?

The feature is covered by two new unit tests.

## Markup:

```ts
createVuetify({
  theme: {
    cspNonce: 'my-csp-nonce',
  },
})
```

<details>

```vue
// Paste your FULL Playground.vue here
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [ ] The PR title is no longer than 64 characters.
- [ ] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Review by johnleider — APPROVED** [review]
lgtm pending @KaelWD 's requested changes

File: packages/vuetify/src/composables/theme.ts [hunk 1]
```diff
@@ -318,6 +320,7 @@ export function createTheme (app: App, options?: ThemeOptions): ThemeInstance {
         const el = document.createElement('style')
         el.type = 'text/css'
         el.id = 'vuetify-theme-stylesheet'
+        if (parsedOptions.cspNonce) el.setAttribute('nonce', parsedOptions.cspNonce)
```
[the remark this answers is not in the document]
Comment (hendric-dev) [reply packages/vuetify/src/composables/theme.ts:326]: Thanks for pointing that out.
I've added it to the head aswell. Can you take a look if this suffices?

**Comment by hendric-dev** [comment]
Any idea why the CI is failing? Seems unrelated to the changed files.

**Comment by MajesticPotatoe** [comment]
> Any idea why the CI is failing? Seems unrelated to the changed files.

failures are because `next` is failing

**Review by KaelWD — APPROVED** [review]

**Pull request merged**